src/main/java/com/example/usermanagement/controller/AuthController.java

```diff
@@ -25,9 +33,25 @@ public ResponseEntity<ResponseDto> register(@Valid @RequestBody CreateUserDto cr
         return new ResponseEntity<>(userService.create(createUserdto), HttpStatus.CREATED);
     }
 
-    @PostMapping("authenticate")
-    public ResponseEntity<AuthTokenResponseDto> authenticate(@Valid
-            @RequestBody LoginDto loginDto) {
-        return new ResponseEntity<>(userService.authenticate(loginDto), HttpStatus.OK);
+    @PostMapping("login")
+    public ResponseEntity<String> authenticate(@Valid
+                                                   @RequestBody LoginDto loginDto) {
+        String tokenResponse = userService.authenticate(loginDto);
+        if (tokenResponse != null) {
+            return new ResponseEntity<>(tokenResponse, HttpStatus.OK);
+        } else {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        }
+    }
+
+    @PostMapping("logout")
+    public ResponseEntity<Response> logout(@RequestBody TokenRequestDto token) {
```

Comment:
ideally an authenticated user will have token in the request header

Reply:
For logging out we need refresh token so a user is providing that in the body to perform the action that is apart from the access token.